src/material/progress-spinner/progress-spinner.ts

```diff
@@ -231,7 +231,9 @@ export class MatProgressSpinner extends _MatProgressSpinnerMixinBase implements
 
   /** The radius of the spinner, adjusted for stroke width. */
   _getCircleRadius() {
-    return (this.diameter - BASE_STROKE_WIDTH) / 2;
+    // In the general case adjust the diameter by the base stroke width.
+    // For small diameters adjust by a diameter percentage so that the radius is greater than 0.
+    return Math.max(this.diameter - BASE_STROKE_WIDTH, this.diameter * 0.09) / 2;
```

Comment:
The smallest currently supported diameter is 11, which gives a circle radius of ½ or approximately (0.09*11)/2. So for diameters <= 10 I maintain the same circle radius ratio by using (0.09 * diameter)/2, instead of subtracting a `BASE_STROKE_WIDTH` of 10.